src/controllers/user/index.js

```diff
@@ -0,0 +1,87 @@
+import { CognitoUser, AuthenticationDetails } from 'amazon-cognito-identity-js'
+import UserPool from './user-pool'
+global.fetch = require('node-fetch')
+
+const authenticate = (Username, Password) =>
+  new Promise((resolve, reject) => {
+    console.log({ Username, Pool: UserPool })
+    const user = new CognitoUser({ Username, Pool: UserPool })
+    const authDetails = new AuthenticationDetails({ Username, Password })
+
+    user.authenticateUser(authDetails, {
+      onSuccess: data => {
+        console.log('onSuccess:', data)
+        resolve(data)
+      },
+
+      onFailure: err => {
+        console.error('onFailure:', err)
+        reject(err)
+      }
+    })
+  })
+
+const addUser = (email, password) => new Promise((resolve, reject) => {
+  UserPool.signUp(email, password, [], null, (err, data) => {
+    if (err) return reject(err)
+    return resolve(data)
+  })
+})
+
+const getSession = () =>
+  new Promise((resolve, reject) => {
+    const user = UserPool.getCurrentUser()
+    if (user) {
+      user.getSession((err, session) => {
+        if (err) {
+          reject(err)
+        } else {
+          resolve(session)
+        }
+      })
+    } else {
+      reject()
+    }
+  })
+
+export const getCurrentUser = async (obj, args, context) => {
+  try {
+    const data = await getSession()
```

Comment:
What do you think if we create a separate method into the Cognito service call getCurrentUser and make it return the email address of the current user (if any)? That way, your main function (for the user controller) would be:
```javascript
export const getCurrentUser = ...
  const currentUserEmail = await cognitoService.getCurrentUser()

  return context.models.user.findOne({ where: { email: currentUserEmail } })
```

Otherwise, your user controller knows too much about Cognito (data structure of the response).